tests/unittests/tests-fmt/tests-fmt.c

```diff
@@ -242,72 +242,126 @@ static void test_fmt_hex_bytes(void)
     TEST_ASSERT_EQUAL_INT(0xAF, val3[2]);
 }
 
+static void test_fmt_u16_hex(void)
+{
+    char out[8] = "zzzzzzzz";
+
+    /* Check return count with null buffer input */
+    TEST_ASSERT_EQUAL_INT(4, fmt_u16_hex(NULL, 0xBEEF));
+    /* always four regardless of value size */
+    TEST_ASSERT_EQUAL_INT(4, fmt_u16_hex(NULL, 0));
+
+    TEST_ASSERT_EQUAL_INT(4, fmt_u16_hex(out, 0));
+    TEST_ASSERT(memcmp(out, "0000zzzz", 8) == 0);
+
+
+    TEST_ASSERT_EQUAL_INT(4, fmt_u16_hex(out, 0xBEEF));
+    TEST_ASSERT(memcmp(out, "BEEFzzzz", 8) == 0);
+}
+
 static void test_fmt_u32_hex(void)
 {
-    char out[12] = "zzzzzzzzzzz";
-    uint32_t val = 3735928559;
+    char out[12] = "zzzzzzzzzzzz";
```

Comment:
"An array of character type may be initialized by a character string literal, optionally
enclosed in braces. Successive characters of the character string literal (including the
terminating null character if there is room or if the array is of unknown size) initialize the
elements of the array" Section 6.7.8 Statement 14 in ISO/IEC 9899:TC3

I think my code is therefor correct and not a single `\0` is stored here?

This might bug @pyropeter's interest :disguised_face:

Reply:
Nice job finding those! No idea how that happen, i double counted them all before committing. 
I think its ready for merge - do you give your blessing?

~~This PR wasn't fun~~ :clown_face: